pkg/operator/startupmonitorreadiness/readiness_checks.go

```diff
@@ -275,11 +276,20 @@ func goodHealthzEtcdEndpoint(client *http.Client, rawURL string) func(context.Co
 func doHTTPCheckAndTransform(ctx context.Context, client *http.Client, rawURL string, checkName string, httpCheckFn func(ctx context.Context, client *http.Client, rawURL string) (int, string, error)) (bool, string, string) {
 	statusCode, response, err := httpCheckFn(ctx, client, rawURL)
 	if err != nil {
+		if utilnet.IsConnectionRefused(err) {
+			return false, "ConnectionRefused", fmt.Sprintf("kube-apiserver is not reachable: %v", err)
```

Comment:
would unify all of these into "APIError".

Then this one: "waiting for kube-apiserver to listen on port :6443"

Reply:
won't `APIError` imply we got an error from the server?

Reply:
network errors are a thing in golang
```
 if netError, ok := err.(net.Error)
```
they also have a Timeout() and Temporary() method to help identify them